src/main/java/hudson/plugins/promoted_builds/inheritance/JobPropertyImplSelector.java

```diff
@@ -0,0 +1,66 @@
+
+package hudson.plugins.promoted_builds.inheritance;
+
+import org.apache.log4j.Logger;
+
+import hudson.Extension;
+import hudson.model.JobProperty;
+
+import hudson.plugins.project_inheritance.projects.InheritanceProject;
+import hudson.plugins.project_inheritance.projects.inheritance.InheritanceSelector;
+
+import hudson.plugins.promoted_builds.JobPropertyImpl;
+
+/**
+ *  
+ * @author Jacek Tomaka
+ */
+@Extension(optional=true)
+public class JobPropertyImplSelector extends InheritanceSelector<JobProperty<?>> {
+	private static final long serialVersionUID = 6297336734737164557L;
+	private static Logger logger = Logger.getLogger(JobPropertyImplSelector.class);
+
+	@Override
+	public boolean isApplicableFor(Class<?> clazz){
+		return JobProperty.class.isAssignableFrom(clazz);
+	}
+	
+	@Override
+	public InheritanceSelector.MODE getModeFor(Class<?> clazz){
+		if (JobPropertyImpl.class.isAssignableFrom(clazz)) return MODE.USE_LAST;
+		return MODE.NOT_RESPONSIBLE;
+	}
+	
+	@Override
+	public String getObjectIdentifier(JobProperty<?> obj){
+		if (JobPropertyImpl.class.getName().equals(obj.getClass().getName())){
+			return "JobPropertyImpl";
+		}
+		return null;
+	}
+	
+	@Override
+	public JobPropertyImpl merge(JobProperty<?> prior, JobProperty<?> latter, InheritanceProject caller){
+		return null;
```

Comment:
From what I see in in https://github.com/i-m-c/jenkins-inheritance-plugin/blob/master/src/main/java/hudson/plugins/project_inheritance/projects/inheritance/InheritanceSelector.java#L258-L268 , it may cause an NPE in your handleSingleton implementation